compiler/rustc_hir_analysis/src/hir_ty_lowering/bounds.rs

```diff
@@ -23,44 +23,203 @@ use crate::hir_ty_lowering::{
     AssocItemQSelf, FeedConstTy, HirTyLowerer, PredicateFilter, RegionInferReason,
 };
 
-impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
-    /// Add a `Sized` bound to the `bounds` if appropriate.
+impl<'tcx, 'a> dyn HirTyLowerer<'tcx> + '_
+where
+    'tcx: 'a,
+{
+    /// Add a `Sized` or `experimental_default_bounds` bounds to the `bounds` if appropriate.
     ///
-    /// Doesn't add the bound if the HIR bounds contain any of `Sized`, `?Sized` or `!Sized`.
-    pub(crate) fn add_sized_bound(
+    /// Doesn't add the bound if the HIR bounds contain any of `Trait`, `?Trait` or `!Trait`.
```

Comment:
move this comment to `add_default_trait`. This function adds all `Sized` and `exper